pageserver/src/tenant/timeline.rs

```diff
@@ -644,18 +646,27 @@ impl Timeline {
     /// # Cancel-Safety
     ///
     /// This method is cancellation-safe.
+    #[inline(always)]
     pub(crate) async fn get(
         &self,
         key: Key,
         lsn: Lsn,
         ctx: &RequestContext,
+    ) -> Result<Bytes, PageReconstructError> {
+        self.timeline_get_throttle.throttle(ctx, 1).await;
+        self.get0(key, lsn, ctx).await
+    }
+    /// Not subject to [`Self::timeline_get_throttle`].
+    async fn get0(
```

Comment:
nit: I'd go for `get_impl` to have it simetrical with `get_vectored_impl`

Reply:
7e5c93fc70b6fd824a64b047aefbd6f8801c48b7